notification-server/src/bin/service.rs

```diff
@@ -329,6 +329,10 @@ async fn traverse_chain(
                 processed_height = block_height;
                 let delta = start.elapsed();
                 histogram!("block.process_successful_duration").record(delta);
+                if let Ok(duration_since_epoch) = SystemTime::now().duration_since(UNIX_EPOCH) {
+                    let current_timestamp = duration_since_epoch.as_secs() as f64;
+                    gauge!("block.process_successful_last_unix").set(current_timestamp);
+                }
```

Comment:
If your goal is to trace when the service is down, you could do it in a more precise and efficient (+ configurable) way I'd say. Something like

```rust
struct Args {
  ...
  alive_timeout_seconds: u8,
}

async fn main() {
  tokio::select! {
    _ = liveness_process(args.alive_timeout_seconds, ...) => {}
    _ = chain_traversal(...) => {}
  }
}
```

Reply:
I did write an invalid title. The purpose is not to determine whether the service is up or down, but instead to trace whether it is successfully processing blocks. If no blocks has been successfully processed within the last say 5m then I will emit an alert. Much like this: 

https://metrics.internal.testnet.concordium.com/d/JWAZ2MiSk/alerts-non-critical?orgId=1&viewPanel=59